lib/test/unit/test-thread-result.rb

```diff
@@ -0,0 +1,51 @@
+#--
+#
+# Author:: Tsutomu Katsube.
+# Copyright:: Copyright (c) 2024 Tsutomu Katsube. All rights reserved.
+# License:: Ruby license.
+
+module Test
+  module Unit
+    class TestThreadResult
```

Comment:
Can we rename this to `ThreadTestResult` or `SubTestResult`?
We should not split "test" and "result".

Reply:
Yes we can. I prefer `SubTestResult` because It's assigned to the `sub_result` variable:

```ruby
sub_result = SubTestResult.new(result)
```

I confirmed that all tests passed, with the summary count matching the sequential run. Thanks!